src/Splat.DryIoc/DryIocDependencyResolver.cs

```diff
@@ -39,16 +40,22 @@ public DryIocDependencyResolver(IContainer container = null)
                 ? _container.ResolveMany(serviceType)
                 : _container.ResolveMany(serviceType, serviceKey: contract);
 
+        /// <inheritdoc />
+        public bool HasRegistration(Type serviceType)
+        {
+            return _container.GetServiceRegistrations().Any(x => x.ServiceType == serviceType);
+        }
+
         /// <inheritdoc />
         public virtual void Register(Func<object> factory, Type serviceType, string contract = null)
         {
             if (string.IsNullOrEmpty(contract))
             {
-                _container.UseInstance(serviceType, factory(), IfAlreadyRegistered.AppendNewImplementation);
+                _container.UseInstance(serviceType, factory(), IfAlreadyRegistered.Replace);
```

Comment:
This won't match the modern dependency resolver I don't think, it just appends to the linked list for that bucket. Not sure if that matters.

Reply:
doh when i was comparing the behaviours, i missed the ! in the modern dependency resolver

`if (!_registry.ContainsKey(pair))`

modern dependency resolver returns last item in the bucket.
1) do we need to hold multiple registrations of things like the ilogger factory? if it's being done by type and contractname?
2) other ioc containers may not return the last item (if they even support multiple registrations) so we have to be aware of nuances. The logging extensions may not work as desired on anything other than modern dependency resolver.

i'll have a think and build some tests up. welcome thoughts as i only use very lightweight use cases of the servicelocator.

Reply:
autofac, dryioc, modern dependency resolver now have same behaviour.

Reply:
There is a method associated with GetAllRegistrations too that might need some testing around for the various DI engines to make sure they work as we expect.

Reply:
https://github.com/reactiveui/splat/blob/6f3381952f3680abb8d718eda974a3d4fd53d052/src/Splat/ServiceLocation/IReadonlyDependencyResolver.cs#L32